### ISSUE TITLE
kiall - segmentation fault - AT_NULL terminator not found, cannot parse auxv structure

### ISSUE THREAD

**Issue by PeterPitterling** [HewlettPackard/LinuxKI#57]
getting 'segfault' on SLES15.3 + SLES12.5 during kiall processing

```console
# /opt/linuxki/kiall -r
Processing files in: /.../0827_1223
Merging KI binary files.  Please wait...
kiinfo -likimerge cannot merge ki.bin files.  Suspect default tracing (ftrace) used.
/opt/linuxki/kiinfo -kitrace -ts 0827_1223
/opt/linuxki/kiinfo -kiall csv -html -ts 0827_1223
/opt/linuxki/kiall: line 278: 24861 Segmentation fault      (core dumped) $DIR/kiinfo -kiall csv$oracle$mangle$blkfrq -html -ts $tag
kiall complete
```

/var/log/messages contains 2 entries - 2 files were also written
```console
kernel: [1233077.597260] kiinfo[24861]: segfault at 7fcad2d94580 ip 0000000000425c30 sp 00007ffcf1299e20 error 4 in kiinfo[400000+e4000]
kernel: [1233077.763743] kiinfo[24854]: segfault at 7f3ad8100580 ip 00000000004a9279 sp 00007ffd3c608c80 error 4 in kiinfo[400000+e4000]
systemd-coredump[24873]: AT_NULL terminator not found, cannot parse auxv structure.
systemd-coredump[24874]: AT_NULL terminator not found, cannot parse auxv structure.
```

any clue?

**Comment by MarkCRay** [comment]
The coredump was due to several of the per-cpu ki.bin files getting truncated such that a trace record was partially logged and the trace file ended.   This is very unusual as there were no error logged and I haven't been able to duplicate the issue.  

However, I have modified kiinfo to detect end of the file in the middle of a trace event so that all the trace events up to that point will be processed.   For example, in this case, there was still 18.5 seconds of good data and 45M trace events that were processed.  

Kiinfo will print the following error when the issue is detected:

```
Warning: Truncated KI binary detected for cpu 2
Proceeding as if end of trace collection was found
```

This will be fixed in the 7.8 version.
